packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -291,14 +290,20 @@ export class DataStores implements IDisposable {
         return this.aliasMap.get(id) !== undefined || this.contexts.get(id) !== undefined;
     }
 
-    public bindFluidDataStore(fluidDataStoreRuntime: IFluidDataStoreChannel): void {
-        const id = fluidDataStoreRuntime.id;
+    /**
+     * Makes the data store with the given id visible in the container graph by moving the data store context from
+     * unbound to bound list. This data store can now be reached from the root.
+     * @param id - The id of the data store context to make visible.
+     */
+    public makeDataStoreVisible(id: string): void {
```

Comment:
Are we connecting globally or locally?

Reply:
Well, the data store calls here when it is ready to be locally visible. It may also become globally visible if the container is attached.
DataStores moves it from unbound to bound list essentially making it visible (or reachable) in this container. Basically, this data store can now be retrieved by calling `getDataStore`.

Reply:
From what I understand, this method makes the datastore locally visible, and global visibility is a side effect of the container being attached. It would be cool if we call this `makeDataStoreLocallyVisible`; however, I'm not sure how we want to indicate this side effect.

Reply:
Also, how much of this logic can we move to the datastore itself? I'd prefer we'd have as little logic here if all we're using it is in the datastore itself.

Reply:
From data store's perspective, its ready to be visible (both locally and globally). Whether it becomes globally visible or not is up to the container runtime and it controls when the data store becomes globally visible.
Ideally I would want the data store to not even care about being globally visible but today it does.